mtp_api/apps/security/views.py

```diff
@@ -39,7 +44,40 @@ def filter(self, qs, value):
         return qs
 
 
+class MonitorProfileMixin(viewsets.GenericViewSet):
+    def get_monitor_object(self):
+        return self.get_object()
+
+    @decorators.action(
+        methods=['post'], detail=True,
+        permission_classes=[IsAuthenticated, NomsOpsClientIDPermissions],
+        url_path='monitor', url_name='monitor'
+    )
+    def monitor(self, request, pk=None):
+        obj = self.get_monitor_object()
+        obj.monitoring_users.add(request.user)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @decorators.action(
+        methods=['post'], detail=True,
+        permission_classes=[IsAuthenticated, NomsOpsClientIDPermissions],
+        url_path='unmonitor', url_name='unmonitor'
+    )
+    def unmonitor(self, request, pk=None):
+        obj = self.get_monitor_object()
+        obj.monitoring_users.remove(request.user)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
 class SenderProfileListFilter(django_filters.FilterSet):
+    sender = MultipleFieldCharFilter(
+        field_name=(
+            'bank_transfer_details__sender_name',
+            'debit_card_details__cardholder_name__name',
+            'debit_card_details__sender_email__email',
+        ),
+        lookup_expr='icontains'
+    )
```

Comment:
Not really sure about this `sender`, it's not obvious that it searches for email address as well and it's quite similar to the `search` field I added in my PR. Also, is it actually used?

Reply:
yup, the purpose of this field was exactly the same as yours. we added it back in may, but as you can see it never got merged :) i'll delete it from this branch